modules/gmc-agent/src/test/java/org/gridgain/agent/AgentSelfTest.java

```diff
@@ -112,16 +86,18 @@ public void shouldSendInitialStates() throws Exception {
      */
     @Test
     public void shouldSendChangedClusterTopology() throws Exception {
-        IgniteClusterEx cluster = startGrid(0).cluster();
+        IgniteEx ignite = startGrid(0);
+        changeGmcUri(ignite);
 
+        IgniteClusterEx cluster = ignite.cluster();
         cluster.active(true);
 
         startGrid(1);
 
         assertWithPoll(
             () -> {
                 TopologySnapshot top = interceptor.getPayload(buildClusterTopologyDest(cluster.id()), TopologySnapshot.class);
-                return top.getNodes().size() == 2;
+                return top != null && top.getNodes().size() == 2;
```

Comment:
how we can get `top == null` ?

Reply:
Data race, when we try to get topology, but agent don't connect to new backend with new port.